canopy_bioemi_mod.F90

```diff
@@ -0,0 +1,310 @@
+                    (/ ATEMP_4_SUN,ATEMP_3_SUN /),ZK(i))
+                BTEMP_SUN(i)   = interp_linear1_internal((/ FCH*(2.0_rk/5.0_rk),FCH*(3.0_rk/5.0_rk) /), &
+                    (/ BTEMP_4_SUN,BTEMP_3_SUN /),ZK(i))
+                ATEMP_SHADE(i) = interp_linear1_internal((/ FCH*(2.0_rk/5.0_rk),FCH*(3.0_rk/5.0_rk) /), &
+                    (/ ATEMP_4_SHADE,ATEMP_3_SHADE /),ZK(i))
+                BTEMP_SHADE(i) = interp_linear1_internal((/ FCH*(2.0_rk/5.0_rk),FCH*(3.0_rk/5.0_rk) /), &
+                    (/ BTEMP_4_SHADE,BTEMP_3_SHADE /),ZK(i))
+            else if (ZK(i) .le. FCH*(2.0_rk/5.0_rk) ) then  !Level 4 - Bottom
+                ATEMP_SUN(i)   = interp_linear1_internal((/ ZK(1),FCH*(2.0_rk/5.0_rk) /), &
+                    (/ ATEMP_5_SUN,ATEMP_4_SUN /),ZK(i))
+                BTEMP_SUN(i)   = interp_linear1_internal((/ ZK(1),FCH*(2.0_rk/5.0_rk) /), &
+                    (/ BTEMP_5_SUN,BTEMP_4_SUN /),ZK(i))
+                ATEMP_SHADE(i) = interp_linear1_internal((/ ZK(1),FCH*(2.0_rk/5.0_rk) /), &
+                    (/ ATEMP_5_SHADE,ATEMP_4_SHADE /),ZK(i))
+                BTEMP_SHADE(i) = interp_linear1_internal((/ ZK(1),FCH*(2.0_rk/5.0_rk) /), &
+                    (/ BTEMP_5_SHADE,BTEMP_4_SHADE /),ZK(i))
+            end if
+        end do
+
+        TLEAF_SUN   = ATEMP_SUN + (BTEMP_SUN*TEMP2)
+        TLEAF_SHADE = ATEMP_SHADE + (BTEMP_SHADE*TEMP2)
+        TLEAF_AVE = (TLEAF_SUN*RJCF) + (TLEAF_SHADE*(1.0-RJCF)) ! average = sum sun and shade weighted by sunlit fraction
+
+! Use exponential PPFD model based on Silva et al. (2020) to get approx. sun/shade PPFD
+! through canopy
+!Citation:
+!Silva, S. J., Heald, C. L., and Guenther, A. B.: Development of a reduced-complexity plant canopy
+!physics surrogate model for use in chemical transport models: a case study with GEOS-Chem v12.3.0,
+!Geosci. Model Dev., 13, 2569–2585, https://doi.org/10.5194/gmd-13-2569-2020, 2020.
+        do i=1, SIZE(ZK)  !calculate linear change in parameters interpolated to Silva et al. 5 layer canopy regions
+            if (ZK(i) .gt. FCH) then ! above canopy, PPFD_leaf = PPFD_toc (toc=top of canopy)
+                CTEMP_SUN(i)   = 0.0
+                DTEMP_SUN(i)   = 0.0
+                CTEMP_SHADE(i) = 0.0
+                DTEMP_SHADE(i) = 0.0
+            else if (ZK(i) .le. FCH .and. ZK(i) .gt. FCH*(4.0_rk/5.0_rk)) then  !Level 1 - 2
+                CTEMP_SUN(i)   = interp_linear1_internal((/ FCH*(4.0_rk/5.0_rk),FCH /), &
+                    (/ CTEMP_2_SUN,CTEMP_1_SUN /),ZK(i))
+                DTEMP_SUN(i)   = interp_linear1_internal((/ FCH*(4.0_rk/5.0_rk),FCH /), &
+                    (/ DTEMP_2_SUN,DTEMP_1_SUN /),ZK(i))
+                CTEMP_SHADE(i) = interp_linear1_internal((/ FCH*(4.0_rk/5.0_rk),FCH /), &
+                    (/ CTEMP_2_SHADE,CTEMP_1_SHADE /),ZK(i))
+                DTEMP_SHADE(i) = interp_linear1_internal((/ FCH*(4.0_rk/5.0_rk),FCH /), &
+                    (/ DTEMP_2_SHADE,DTEMP_1_SHADE /),ZK(i))
```

Comment:
this part is a bit ugly, could instead set some local variables, for what to interp and then do the interp outside the loop

Reply:
@zmoon I tend to agree with you, but gets the job done.  Please go ahead and make suggested changes and we can commit and test.  Thanks!

Reply:
Since this is internal, think I will hold off on this, we can always come back to it.